docs/DeveloperGuide.md

```diff
@@ -5,9 +5,262 @@
+
+> Note that `Parser-getPatternMatcher(regex, input, paramFlags)` will be called as long as the command word in the user input is valid,
+> even if the parameters are invalid. The `Matcher` object returned will try to match the input parameters to the given regex.
+> A `TrackerException` will be thrown if the input parameters are invalid.
+
+The input parameters will be passed into another method, `Parser-makeStringPattern(inputParam, paramFlags)` that returns a string in a pattern
+according to a preset format that contains all the necessary input parameters.
+Converting the parameter string to a pattern is done because **order does not matter** in the user's input parameters _(if there are multiple parameters)._
+Thus, it would be necessary to convert the user's input into a consistent format for pattern matching using regex.
+
+> Note: 
+> - All command regex are in the format:\
+>   `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) ` with `a/`, `b/`, `c/` being the respective parameter flags
+>   and `/` as the flag character.
+>   - As of now, parameter flags must only contain 1 character right before the `/`
+>   - Suppose we have `a/aaa b/bbb c/ccc ` as an input, according to the regex above, 
+>   aaa will be in the named capture group "grp1", bbb in "grp2", ccc in "grp3"
+> - If a parameter is optional, we would have it in a non-capture group with a `?` at the end
+>   - i.e. in `a/(?<grp1>.*) (?<grp2>(?:b/.*)?)`, the `b/` flag is an optional parameter
+> 
+> For example, suppose the program wants to match the user input parameter
+> `c/coconut a/apple b/bear a/anaconda d/donkey` to the regex `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) (?<grp4>(?:e/.*)?)`
+> 1. The input parameter string and a string array of flags `{a, b, c, e}` is passed into `Parser-makeStringPattern(...)`
+> 2. `Parser-makeStringPattern` will return a pattern string `a/apple b/bear c/coconut `
+> 3. The program will then try to match the pattern string `a/apple b/bear c/coconut ` to the regex `
+> a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) (?<grp4>(?:e/.*)?)`
+> 4. The above pattern string matches the regex, with the following strings in the named capture groups:
+>    - "grp1": apple
+>    - "grp2": bear
+>    - "grp3": coconut
+>    - "grp4" is empty
+
+The `Matcher` will detect whether this input string pattern match the given regex pattern given to it
+and extract out the necessary information if there is a match. 
+This will be used by each command's respective parsing method and returns the relevant parsed `Command`
+object to `SuperTracker+handleCommands()`
+
+### File Saving and Loading
+Saving and loading data of `Item` objects is performed by the `FileManager` class. As the program is running, 
+`Item` objects will be stored in the `Inventory` class. The following is a class diagram of `FileManager`
+and its relevant dependencies\
+![FileManagerClass](./uml-diagrams/FileManagerClass.png)
```

Comment:
Class diagram is too complicated, can remove redundant attributes or methods to make it easier to understand for this implementation.